platforms/jvm/testing-junit-platform/src/main/java/org/gradle/api/internal/tasks/testing/junitplatform/JUnitPlatformTestExecutionListener.java

```diff
@@ -124,6 +125,7 @@ public void executionStarted(TestIdentifier testIdentifier) {
     @Override
     public void executionFinished(TestIdentifier testIdentifier, TestExecutionResult testExecutionResult) {
         if (testExecutionResult.getStatus() == ABORTED) {
+            testExecutionResult.getThrowable().ifPresent(throwable -> resultProcessor.failure(getId(testIdentifier), DefaultTestFailure.fromTestAssumptionFailure(throwable)));
```

Comment:
❓ I find this if block a little confusing.  We're checking if we need to report a failure (from an assumption), and then in the next line below, immediately report skipped?  It seems to me that only one of these should be needed - why does the TestNG adapter not need to call `reportSkipped` after determining an assertion failure?